flixel/addons/effects/chainable/FlxOutlineEffect.hx

```diff
@@ -66,44 +96,127 @@ class FlxOutlineEffect implements IFlxEffect
 	{
 		if (dirty)
 		{
-			var brush = (thickness * 2) + 1;
+			var brush:Int = (thickness * 2);
 			
 			if (_pixels == null || _pixels.width < bitmapData.width + brush || _pixels.height < bitmapData.height + brush)
 			{
-				_pixels = new BitmapData(bitmapData.width + brush, bitmapData.height + brush, true, FlxColor.TRANSPARENT);
+				_pixels = new BitmapData(bitmapData.width + brush, bitmapData.height + brush, true, color);
 			}
 			else
 			{
-				_pixels.fillRect(_pixels.rect, FlxColor.TRANSPARENT);
+				_pixels.fillRect(_pixels.rect, color);
 			}
 			
-			for (y in 0...bitmapData.height)
+			if (mode == PIXEL_BY_PIXEL)
 			{
-				for (x in 0...bitmapData.width)
+				_pixels.fillRect(_pixels.rect, FlxColor.TRANSPARENT);
+				
+				for (y in 0...bitmapData.height)
 				{
-					var c:FlxColor = bitmapData.getPixel32(x, y);
-					if (c.alpha > threshold)
+					for (x in 0...bitmapData.width)
 					{
-						surroundPixel(x, y, brush);
+						var c:FlxColor = bitmapData.getPixel32(x, y);
+						if (c.alpha > threshold)
+						{
+							surroundPixel(x, y, brush);
+						}
 					}
 				}
 			}
+			else
```

Comment:
`apply()` is getting too long for my taste, splitting it up into smaller functions would make it more readable.